src/Features/Core/Portable/CodeRefactorings/MoveType/AbstractMoveTypeService.Editor.cs

```diff
@@ -0,0 +1,272 @@
+
+                var membersToRemove = GetMembersToRemove(root, typeNode, removeDescendentsOfSelf: _makeTypePartial);
+                foreach (var member in membersToRemove)
+                {
+                    documentEditor.RemoveNode(member, SyntaxRemoveOptions.KeepNoTrivia);
+                }
+
+                var memberToAdd = GetMemberToAdd(documentEditor.Generator, newDocumentName);
+                if (memberToAdd != null)
+                {
+                    documentEditor.AddMember(typeNode, memberToAdd);
+                }
+
+                var modifiedRoot = documentEditor.GetChangedRoot();
+
+                // add an empty document to solution, so that we'll have options from the right context.
+                var solutionWithNewDocument = projectToBeUpdated.Solution.AddDocument(newDocumentId, newDocumentName, string.Empty/*, folders, fullFilePath*/);
+
+                // update the text for the new document
+                solutionWithNewDocument = solutionWithNewDocument.WithDocumentSyntaxRoot(newDocumentId, modifiedRoot, PreservationMode.PreserveIdentity);
+
+                // get the updated document, perform clean up like remove unused usings.
+                var newDocument = solutionWithNewDocument.GetDocument(newDocumentId);
+                return await CleanUpDocumentAsync(newDocument, cancellationToken).ConfigureAwait(false);
+            }
+
+            private async Task<Solution> UpdateSourceDocumentAsync(
+                Document sourceDocument, TTypeDeclarationSyntax typeNode, CancellationToken cancellationToken)
+            {
+                var documentEditor = await DocumentEditor.CreateAsync(sourceDocument, cancellationToken).ConfigureAwait(false);
+
+                AddPartialModifiersToTypeChain(documentEditor, typeNode);
+
+                if (!_makeTypePartial)
+                {
+                    documentEditor.RemoveNode(typeNode, SyntaxRemoveOptions.KeepNoTrivia);
+                }
+
+                var updatedDocument = documentEditor.GetChangedDocument();
+                // TODO: make this optional -- shouldn't touch other parts of source document unless asked to. 
+                return await CleanUpDocumentAsync(updatedDocument, cancellationToken).ConfigureAwait(false);
+            }
+
+            private SyntaxNode GetMemberToAdd(SyntaxGenerator generator, string newDocumentName)
+            {
+                if (!_makeOuterTypesPartial)
+                {
+                    return null;
+                }
+
+                // we recognize filenames of that follow the pattern "OuterClass.InnerClass.cs" 
+                // and generate code for this file, which will look like so:
+                // a partial definition of OuterClass and a nested private InnerClass.
+                var nameParts = Path.GetFileNameWithoutExtension(newDocumentName).Split(new char[] { '.' });
+                if (nameParts == null || nameParts.Count() != 2)
```

Comment:
it's not just Outer.Inner.cs   It could be Outer.Inner1.Inner2.cs etc.  i.e. we shouldn't just be checking for 2